spring-data-mongodb/src/main/java/org/springframework/data/mongodb/util/json/ParameterBindingContext.java

```diff
@@ -67,7 +67,7 @@ public ParameterBindingContext(ValueProvider valueProvider, ExpressionParser exp
 	 * @param expressionEvaluator
 	 * @since 3.1
 	 */
-	public ParameterBindingContext(ValueProvider valueProvider, SpELExpressionEvaluator expressionEvaluator) {
+	public ParameterBindingContext(ValueProvider valueProvider, ValueExpressionEvaluator expressionEvaluator) {
```

Comment:
can we deprecate the current public method and delegate with limited functionality?
```java
@Deprecated(since = "3.3")
public ParameterBindingContext(ValueProvider valueProvider, SpELExpressionEvaluator expressionEvaluator) {
	this(valueProvider, (ValueExpressionEvaluator) expressionEvaluator::evaluate);
}

public ParameterBindingContext(ValueProvider valueProvider, ValueExpressionEvaluator expressionEvaluator) {
...
```